### PR TITLE
Linting (including Karma file indent);

### PR DESCRIPTION
**Pull request by brettz9**
Babelify/Uglify/ESLint and add source maps for test worker;
    avoid polyfill double-loading;
    put test worker in test directory;
    Grunt: Reorder tasks to be more in normal order of execution;
    Load polyfill in worker for sake of PhantomJS tests;
    Overcome or workaround Karma/PhantomJS issues with workers, documenting limitations;
    remove ecmaFeatures:modules (not needed);
    allow override of Saucekey env var.
    rename web workers file to enable to execute first and avoid PhantomJS interaction problem with server-handlers.js;
    promise rejection for bad schema callback

This looks like a lot, but much is linting with whitespace fixes and some preparatory work for my plan to add a PR for versioning to schemas (via idb-schema). But the more substantive aspects here are getting all test files to run on PhantomJS with some working around, the Saucekey env. var. behavior we discussed, and rejecting a promise upon a bad schema callback

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
-module.exports = function(grunt) {
+module.exports = function (grunt) {
     'use strict';
     // Project configuration.
-    var saucekey = process.env.saucekey;
+    var saucekey = process.env.SAUCE_ACCESS_KEY || process.env.saucekey;
```
Comment (aaronpowell) [inline Gruntfile.js]: I'll remove the settings from the `yml` file so this won't be needed, can you revert this particular change (and the one on line 7)?

**Comment by brettz9** [comment]
I updated...I assume that's what you wanted.

**Comment by aaronpowell** [comment]
Yep! I want to try running this PR locally, so I'll (hopefully) get to it today, if not it'll be this week

**Comment by aaronpowell** [comment]
So what's the role of browserify in this now?

**Comment by brettz9** [comment]
Sorry, I had been using browserify here to include polyfills by `import` statements, but it managed to work out better by using `importScripts` inside the worker instead. So I can remove it from this particular PR if you like, but I'm planning to shortly include another PR which imports [`idb-schema`](https://github.com/treojs/idb-schema) using ES6 modules syntax, but which relies upon `require` loading under the hood, so browserify is needed for that.

**Pull request merged**